app/models/concerns/postcode_normalize.rb

```diff
@@ -0,0 +1,12 @@
+module PostcodeNormalize
+  extend ActiveSupport::Concern
+  included do
+    def postcode=(str)
+      if str
+        super UKPostcode.parse(str).to_s
+      else
+        super str
```

Comment:
MINOR: It looks like we're guarding against running `UKPostcode.parse(nil)` here, but it's not tested below. How important is it to guard against this and is it worth testing?

Reply:
Explained by @tvararu: if we don't guard against `nil` here then many tests fail, so it's tested indirectly. LGTM.

Reply:
I added a unit test anyway